Lib/Inc/adc_data.h

```diff
@@ -13,21 +13,27 @@
 #include "adc.h"
 #include "log.h"
 
-class CAdc
+/**
+ *  @brief Number of active ADC Channels,
+ *   change this according to hardware
+ */
+#define ADC_CHANNELS 14
+
+class CAdcData
 {
 public:
-    CAdc(ADC_HandleTypeDef *p_hadc, uint8_t adc_channels);
-    ~CAdc();
+    CAdcData(ADC_HandleTypeDef *p_hadc);
 
     // Public methods
+    void init();
     // use [] operator to read a particular ADC Channel
     uint16_t operator[](uint8_t adc_channel);
-    void triggerAdc();
+    void triggerRead();
```

Comment:
meh. I am still not liking the name. Just looks nice. But this is just a matter of taste in this case. You generally make decent choices as far as names go.
```
adc_data.trigger();
```